lib/storage/authgateway.go

```diff
@@ -422,80 +451,128 @@ func (gw *AuthGatewayV2) GetWebPublicAddrs() (addrs []string) {
 }
 
 // SetWebPublicAddrs sets proxy web service public addresses.
-func (gw *AuthGatewayV2) SetWebPublicAddrs(value []string) {
+func (gw *AuthGatewayV1) SetWebPublicAddrs(value []string) {
 	gw.Spec.WebPublicAddr = &value
 }
 
 // CheckAndSetDefaults validates the resource and fills in some defaults.
-func (gw *AuthGatewayV2) CheckAndSetDefaults() error {
+func (gw *AuthGatewayV1) CheckAndSetDefaults() error {
 	if gw.Metadata.Name == "" {
 		gw.Metadata.Name = KindAuthGateway
 	}
 	err := gw.Metadata.CheckAndSetDefaults()
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	if gw.Spec.ConnectionLimits != nil {
```

Comment:
nit: If you implemented Check on the pointer receiver you could check for nil inside the implementation.